hwilib/cli.py

```diff
@@ -204,38 +201,19 @@ def process_commands(cli_args):
         if not client:
             return {'error':'Could not find device with specified fingerprint','code':DEVICE_CONN_ERROR}
     elif args.device_type and args.device_path:
-        try:
+        with handle_errors(result=result, code=DEVICE_CONN_ERROR):
             client = get_client(device_type, device_path, password)
-        except HWWError as e:
-            return {'error': e.get_msg(), 'code': e.get_code()}
-        except Exception as e:
-            return {'error':str(e),'code':DEVICE_CONN_ERROR}
+        if 'error' in result:
+            return result
     else:
         return {'error':'You must specify a device type or fingerprint for all commands except enumerate','code':NO_DEVICE_PATH}
 
     client.is_testnet = args.testnet
 
     # Do the commands
-    try:
+    with handle_errors(result=result, debug=args.debug):
         result = args.func(args, client)
-    except HWWError as e:
-        result = {'error': e.get_msg(), 'code': e.get_code()}
-    except Exception as e:
-        if args.debug:
-            import traceback
-            traceback.print_exc()
-        result = {'error': str(e), 'code': UNKNOWN_ERROR}
-
-    # Close the device
-    try:
         client.close()
```

Comment:
This still needs to be in its own `with` block. Otherwise if there's an error, the client won't be closed which can cause issues with using the device later.

Reply:
ah right, good call